vars/cloudFoundryCreateServiceKey.groovy

```diff
@@ -0,0 +1,118 @@
+
+@Field Set STEP_CONFIG_KEYS = [
+    'cloudFoundry',
+        /**
+         * Cloud Foundry API endpoint.
+         * @parentConfigKey cloudFoundry
+         */
+        'apiEndpoint',
+        /**
+         * Cloud Foundry credentials.
+         * @parentConfigKey cloudFoundry
+         */
+        'credentialsId',
+        /**
+         * Cloud Foundry target organization.
+         * @parentConfigKey cloudFoundry
+         */
+        'org',
+        /**
+         * Cloud Foundry target space.
+         * @parentConfigKey cloudFoundry
+         */
+        'space',
+        /**
+         * Cloud Foundry service instance, for which the service key will be created.
+         * @parentConfigKey cloudFoundry
+         */
+        'serviceInstance',
+        /**
+         * Cloud Foundry service key, which will be created.
+         * @parentConfigKey cloudFoundry
+         */
+        'serviceKey',
+        /**
+         * Cloud Foundry service key configuration.
+         * @parentConfigKey cloudFoundry
+         */
+        'serviceKeyConfig',
+    /** @see dockerExecute */
+    'dockerImage',
+    /** @see dockerExecute */
+    'dockerWorkspace'
+]
+
+@Field Set GENERAL_CONFIG_KEYS = STEP_CONFIG_KEYS
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * Step that creates a service key for a service instancve on Cloud Foundry
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = checkScript(this, parameters) ?: this
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('cloudFoundry/org')
+            .withMandatoryProperty('cloudFoundry/space')
+            .withMandatoryProperty('cloudFoundry/credentialsId')
+            .withMandatoryProperty('cloudFoundry/serviceInstance')
+            .withMandatoryProperty('cloudFoundry/serviceKey')
+            .use()
+
+        echo "[${STEP_NAME}] Info: docker image: ${config.dockerImage}, docker workspace: ${config.dockerWorkspace}"
+        executeCreateServiceKey(script, config)
+    }
+}
+
+private def executeCreateServiceKey(script, Map config) {
+    dockerExecute(script:script, dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace) {
+
+        withCredentials([
+            usernamePassword(credentialsId: config.cloudFoundry.credentialsId, passwordVariable: 'CF_PASSWORD', usernameVariable: 'CF_USERNAME')
+        ]) {
+            if (config.cloudFoundry.serviceKeyConfig == null) {
+                bashScript =
```

Comment:
Most of the script in if and else block are the same. Consider refactoring in order to avoid code duplications

Reply:
done